### PR TITLE
Map integration

### PR DESCRIPTION
**Pull request by oolusina**
Small steps towards the bigger goal but progress!
image
image

### CODE REVIEW

File: src/main/webapp/index.html [hunk 1]
```diff
@@ -20,6 +20,7 @@
     <script src="https://kit.fontawesome.com/de2fa200e7.js" crossorigin="anonymous"></script>
     </link>
     <script src="script.js"></script>
+    <script src="https://maps.googleapis.com/maps/api/js?key=MAP_API_KEY"></script>
```
Comment (hossam-zaki) [inline src/main/webapp/index.html:23]: won't we need an api key here?

File: src/main/webapp/script.js [hunk 2]
```diff
@@ -12,6 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+let map;
+/* Editable marker that displays when a user clicks in the map. */
+let editMarker;
+
```
Comment (hossam-zaki) [inline src/main/webapp/script.js:18]: usually with js there's a problem with ```let``` so it'd be better to use ```var```

File: src/main/webapp/index.html (hunk 1 above)
Comment (oolusina) [reply src/main/webapp/index.html:23]: Yeah but it’s not safe to upload it to github. If you want to test locally you could just put your own MAP_API Key or I could send mine.

File: src/main/webapp/script.js (hunk 2 above)
Comment (oolusina) [reply src/main/webapp/script.js:18]: This is what they had in their example but I might change it

File: src/main/webapp/index.html (hunk 1 above)
Comment (hossam-zaki) [reply src/main/webapp/index.html:23]: ah ok i see i see. I think charles put on though, so we should be wary of that

File: src/main/webapp/script.js (hunk 2 above)
Comment (hossam-zaki) [reply src/main/webapp/script.js:18]: i'm not 100% sure with let vs var, but if the example has let and it works, then carry on

File: src/main/webapp/script.js [hunk 3]
```diff
@@ -45,6 +45,13 @@ window.onload = function () {
     document.getElementById("Options").remove();
     document.getElementById("added").remove();
   };
+
+  var map_btn = document.getElementById("map_survey");
+  map_btn.onclick = function () {
+    map = new google.maps.Map(
+      document.getElementById('map'),
+      {center: {lat: 38.5949, lng: -94.8923}, zoom: 4});
```
[the remark this answers is not in the document]
Comment (oolusina) [reply src/main/webapp/script.js:59]: Will change in the next PR

File: src/main/webapp/script.js [hunk 4]
```diff
@@ -48,3 +77,44 @@ function getInput() {
...
+  google.maps.event.addListener(infoWindow, 'closeclick', () => {
+    editMarker.setMap(null);
+  });
+
+  infoWindow.open(map, editMarker);
+}
+
+//Creates an editable textbox and a submit button for marker
+function buildInfoWindowInput(lat, lng) {
+  const textBox = document.createElement('textarea');
```
[the remark this answers is not in the document]
Comment (oolusina) [reply src/main/webapp/script.js:104]: Was saving it as a placeholder for next time one working

**Pull request merged**